src/syscalls.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * Enhanced Seccomp x86_64 Syscall Table
+ *
+ * Copyright (c) 2012, 2020 Red Hat <pmoore@redhat.com>
+ * Author: Paul Moore <paul@paul-moore.com>
+ * gperf support: Giuseppe Scrivano <gscrivan@redhat.com>
+ */
+
+#ifndef _SYSCALLS_H
+#define _SYSCALLS_H
+
+#include <stddef.h>
+
+#include "arch-aarch64.h"
+#include "arch-arm.h"
+#include "arch.h"
+#include "arch-mips64.h"
+#include "arch-mips64n32.h"
+#include "arch-mips.h"
+#include "arch-parisc.h"
+#include "arch-ppc64.h"
+#include "arch-ppc.h"
+#include "arch-s390.h"
+#include "arch-s390x.h"
+#include "arch-x32.h"
+#include "arch-x86_64.h"
+#include "arch-x86.h"
+
+struct arch_syscall_def_internal {
+	int name;
+	int index;
+	/* Each arch listed here must be defined in syscalls.c  */
+	int x86_64;
+	int arm;
+	int aarch64;
+	int mips64n32;
+	int mips64;
+	int mips;
+	int parisc;
+	int ppc64;
+	int ppc;
+	int s390;
+	int s390x;
+	int x32;
+	int x86;
+};
+
+#define __SCMP_NR_OABI_SYSCALL_BASE    0x900000
+#define __SCMP_ARM_NR_BASE             0x0f0000
+/* NOTE: we currently only support the ARM EABI, more info at the URL below:
+ *       -> http://wiki.embeddedarm.com/wiki/EABI_vs_OABI */
+#if 1
+#define __SCMP_NR_BASE			0
+#else
+#define __SCMP_NR_BASE			__SCMP_NR_OABI_SYSCALL_BASE
+#endif
```

Comment:
If we are going to move all of the per-ABI offsets into one file, we need to make sure the offset names are tied to their associated ABI.  As an example, the name `__SCMP_NR_BASE` is not a good choice here as the associated ABI is ambigious, but a name like `__SCMP_ARM_NR_BASE` is better as it clearly ties the offset define to the ARM ABI.

Does that make sense?